pkg/wakaama/contrib/lwm2m_objects/device/device.c

```diff
@@ -0,0 +1,357 @@
+    }
+
+    /* Full object requested */
+    if (*num_dataP == 0) {
+        /* This list must contain all readable resources */
+        uint16_t resList[] = {
+            LWM2M_RES_MANUFACTURER, LWM2M_RES_MODEL_NO,   LWM2M_RES_SERIAL,
+            LWM2M_RES_FW_VER,       LWM2M_RES_HW_VERSION, LWM2M_RES_SW_VERSION,
+            LWM2M_RES_BINDINGS,     LWM2M_RES_TYPE,       LWM2M_RES_ERROR_CODE,
+        };
+        int cnt = sizeof(resList) / sizeof(uint16_t);
+        *data_arrayP = lwm2m_data_new(cnt);
+        if (*data_arrayP == NULL) {
+            return COAP_500_INTERNAL_SERVER_ERROR;
+        }
+        *num_dataP = cnt;
+        for (i = 0; i < cnt; i++) {
+            (*data_arrayP)[i].id = resList[i];
+        }
+    }
+
+    for (i = 0; i < *num_dataP; i++) {
+        switch ((*data_arrayP)[i].id) {
+            /* Exec resources */
+            case LWM2M_RES_REBOOT:
+            case LWM2M_RES_FRESET:
+            case LWM2M_RES_ERROR_CODE_RESET:
+                result = COAP_405_METHOD_NOT_ALLOWED;
+                break;
+            /* Statically defined resources */
+            case LWM2M_RES_MANUFACTURER:
+                lwm2m_data_encode_string(LWM2M_DEVICE_MANUFACTURER,
+                                         *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_MODEL_NO:
+                lwm2m_data_encode_string(LWM2M_DEVICE_MODEL, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_SERIAL:
+                lwm2m_data_encode_string(LWM2M_DEVICE_SERIAL, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_FW_VER:
+                lwm2m_data_encode_string(LWM2M_DEVICE_FW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_HW_VERSION:
+                lwm2m_data_encode_string(LWM2M_DEVICE_HW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_SW_VERSION:
+                lwm2m_data_encode_string(LWM2M_DEVICE_SW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_BINDINGS:
+                lwm2m_data_encode_string(LWM2M_DEVICE_BINDINGS, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_TYPE:
+                lwm2m_data_encode_string(LWM2M_DEVICE_TYPE, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            /* Error code is mandatory */
+            case LWM2M_RES_ERROR_CODE:
+                /* TODO: Do this properly */
+                lwm2m_data_encode_int(0, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            /* Time resources */
+            case LWM2M_RES_TIME:
```

Comment:
Why not delete all these cases and leave the default?

Reply:
Will change